src/storage-controller/src/lib.rs

```diff
@@ -1164,6 +1168,40 @@ where
         Ok(())
     }
 
+    fn drop_tables(
```

Comment:
I have mixed feelings about having a separate method for this vs combining this method with `drop_sources`. So let me know if you have a strong opinion.